fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/WorldRenderEvents.java

```diff
@@ -0,0 +1,268 @@
+	 * block entity rendering begins.
+	 *
+	 * <p>Use for global block entity render setup, or
+	 * to append block-related quads to the entity consumers using the
+	 * {@VertexConsumerProvider} from the provided context. This
+	 * will generally give better (if not perfect) results
+	 * for non-terrain translucency vs. drawing directly later on.
+	 */
+	public static final Event<AfterEntities> AFTER_ENTITIES = EventFactory.createArrayBacked(AfterEntities.class, callbacks -> context -> {
+		for (final AfterEntities callback : callbacks) {
+			callback.afterEntities(context);
+		}
+	});
+
+	/**
+	 * Called before default block outline rendering and before checks are
+	 * done to determine if it should happen. Can optionally cancel the default
+	 * rendering but all event handlers will always be called.
+	 *
+	 * <p>Use this to decorate or replace the default block outline rendering
+	 * for specific modded blocks or when the need for a block outline render
+	 * would not be detected.  Normally, outline rendering will not happen for
+	 * entities or other game objects that do not register a block-type hit.
+	 *
+	 * <p>Canceling the default block outline render has no effect on other
+	 * event subscribers - all subscribers will always be called. Generally, modded
+	 * block outline renders are specific to that mod's content or additive
+	 * and thus should not interfere with each other.
+	 *
+	 * <p>This event should NOT be used for general-purpose replacement of
+	 * the default block outline rendering because it will interfere with mod-specific
+	 * renders.  Mods that replace the default block outline should instead
+	 * subscribe to {@link #BLOCK_OUTLINE}.
+	 */
+	public static final Event<BeforeBlockOutline> BEFORE_BLOCK_OUTLINE = EventFactory.createArrayBacked(BeforeBlockOutline.class, callbacks -> context -> {
+		for (final BeforeBlockOutline callback : callbacks) {
+			callback.beforeBlockOutline(context);
+		}
+	});
+
+	/**
+	 * Called after block outline render checks are made and before the
+	 * default block outline render runs.  Will NOT be called if the default outline
+	 * render was cancelled in {@link #BEFORE_BLOCK_OUTLINE}.
+	 *
+	 * <p>Use this to replace the default block outline rendering entirely.
+	 * Canceling the default outline render here avoids interfering
+	 * with mod-specific outline renders that cannot be well-handled by a
+	 * general-purpose outline renderer.
+	 *
+	 * <p>Default block outline rendering does not have to happen here.  If a custom
+	 * default outline effect benefits from being drawn at a later stage it can
+	 * be cancelled here and then drawn in {@link #AFTER_TRANSLUCENT} or {@link #LAST}
+	 * if desired. To facilitate this pattern, those events can check
+	 * {@link WorldRenderContext.AfterBlockOutline#didCancelDefaultBlockOutline()}.
+	 */
+	public static final Event<BlockOutline> BLOCK_OUTLINE = EventFactory.createArrayBacked(BlockOutline.class, callbacks -> context -> {
```

Comment:
AfterBlockOutline?

Reply:
Changed to referr to the context instead of the interface.